custom_enumerable_methods.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    my_self = []
+    i = 0
+    while i < my_self.size
+      yield(my_self[i])
+      i += 1
+    end
+    my_self
+  end
```

Comment:
```rb
2.6.5 :007 > array
 => [5, 4, 4, 0, 5, 0, 3, 4, 0, 0, 2, 6, 5, 6, 5, 6, 2, 1, 3, 0, 8, 0, 0, 1, 7, 1, 3, 5, 6, 1, 1, 1, 6, 5, 7, 7, 2, 0, 3, 1, 7, 0, 1, 4, 5, 7, 6, 5, 4, 3, 6, 5, 4, 0, 6, 2, 7, 2, 5, 6, 4, 7, 1, 3, 6, 3, 3, 2, 6, 1, 7, 7, 8, 6, 1, 3, 5, 0, 4, 1, 0, 5, 4, 2, 5, 2, 5, 6, 6, 0, 1, 8, 4, 5, 4, 0, 4, 0, 8, 6] 
2.6.5 :008 > block = proc { |num| num < 4 }
2.6.5 :009 > words = %w[dog door rod blade]
2.6.5 :010 > 
2.6.5 :011 > my_each_output = ''
2.6.5 :012 > block = proc { |num| my_each_output += num.to_s }
2.6.5 :013 > array.each(&block)
 => [5, 4, 4, 0, 5, 0, 3, 4, 0, 0, 2, 6, 5, 6, 5, 6, 2, 1, 3, 0, 8, 0, 0, 1, 7, 1, 3, 5, 6, 1, 1, 1, 6, 5, 7, 7, 2, 0, 3, 1, 7, 0, 1, 4, 5, 7, 6, 5, 4, 3, 6, 5, 4, 0, 6, 2, 7, 2, 5, 6, 4, 7, 1, 3, 6, 3, 3, 2, 6, 1, 7, 7, 8, 6, 1, 3, 5, 0, 4, 1, 0, 5, 4, 2, 5, 2, 5, 6, 6, 0, 1, 8, 4, 5, 4, 0, 4, 0, 8, 6] 
2.6.5 :014 > each_output = my_each_output.dup
2.6.5 :015 > my_each_output = ''
2.6.5 :016 > 
2.6.5 :017 > array.my_each(&block)
 => [] 
2.6.5 :018 > my_each_output == each_output
 => false 
```

Reply:
```rb
2.6.5 :023 > block = proc { |num| num < 4 }
2.6.5 :024 > array.my_each(&block)
 => [] 
2.6.5 :025 > array.each(&block)
 => [5, 4, 4, 0, 5, 0, 3, 4, 0, 0, 2, 6, 5, 6, 5, 6, 2, 1, 3, 0, 8, 0, 0, 1, 7, 1, 3, 5, 6, 1, 1, 1, 6, 5, 7, 7, 2, 0, 3, 1, 7, 0, 1, 4, 5, 7, 6, 5, 4, 3, 6, 5, 4, 0, 6, 2, 7, 2, 5, 6, 4, 7, 1, 3, 6, 3, 3, 2, 6, 1, 7, 7, 8, 6, 1, 3, 5, 0, 4, 1, 0, 5, 4, 2, 5, 2, 5, 6, 6, 0, 1, 8, 4, 5, 4, 0, 4, 0, 8, 6] 
2.6.5 :026 >
```

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more and fix the things.